plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -39,11 +39,30 @@ type KeystoneAuthenticator struct {
 
 // AuthenticatePassword checks the username, password via keystone call
 func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(username string, password string) (user.Info, bool, error) {
-	opts := gophercloud.AuthOptions{
+	var opts gophercloud.AuthOptions
+	opts = gophercloud.AuthOptions{
 		IdentityEndpoint: keystoneAuthenticator.authURL,
 		Username:         username,
 		Password:         password,
 	}
+	if strings.HasSuffix(keystoneAuthenticator.authURL, "/v3") {
+		// username has a format of domain\name.
+		firstIndex := strings.Index(username, "\\")
```

Comment:
@liggitt I completely agree that the client should do it. But as long as it doesn't (which might be a while), I support this change as a temporary solution. After all, the evaluation of pros and cons is pretty good:
+  no impact on existing functionality as v3 is completely unsupported now
+ support for v3
- does not work for domain names with a backslash
- architecturally not perfect